bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java

```diff
@@ -97,14 +96,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             delayRemainingMillis);
                 }
                 // Compare by reference to check if the future changed
-                if (prevFuture == newFuture) {
+                if (isSameFuture(prevFuture, newFuture)) {
                     logger.trace("REFRESH received. Previous refresh ongoing, will wait for it to complete in {} ms",
                             lastRefreshMillis + REFRESH_THROTTLE_MILLIS - System.currentTimeMillis());
                 }
             }
         }
     }
 
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
```

Comment:
Now isolated here, but still warning:
```
[WARNING] AbstractWeatherHandler.java:[107,23] Unsupported @SuppressWarnings("PMD.CompareObjectsWithEquals")
```

Only needed for trace logging, but didn't want to change this. Also did not want to risk refactoring `AtomicReference` logic for this reason.